zipline/data/data_portal.py

```diff
@@ -924,22 +924,29 @@ def get_history_window(self,
         # forward-fill price
         if field == "price":
             if frequency == "1m":
-                data_frequency = 'minute'
+                ffill_data_frequency = 'minute'
             elif frequency == "1d":
-                data_frequency = 'daily'
+                ffill_data_frequency = 'daily'
             else:
                 raise Exception(
                     "Only 1d and 1m are supported for forward-filling.")
 
             assets_with_leading_nan = np.where(isnull(df.iloc[0]))[0]
 
             history_start, history_end = df.index[[0, -1]]
+            if ffill_data_frequency == 'daily' != data_frequency:
+                # When we're looking for a daily value, but we haven't seen any
+                # volume in today's minute bars yet, we need to use the
+                # previous day's ffilled daily price. Using today's daily price
+                # could yield a value from later today.
+                history_start -= pd.Timedelta(days=1)
```

Comment:
Is there a reason this isn't a trading calendar day?

Reply:
Since the most recent close as of the weekend is the same as the most recent close as of Friday, I didn't bother to figure out which trading calendar to use here, especially if there are multiple assets. Do you think this breaks under certain conditions?

Reply:
I don't think it breaks here.

Where it would be relevant is if we had a calendar with a session spanning multiple days, but in practice, I do not think we have that concern.